lib/omniauth/strategies/uber.rb

```diff
@@ -31,12 +31,26 @@ class Uber < OmniAuth::Strategies::OAuth2
         }
       end
 
+      def callback_url
+        # NOTE: This method is originally defined in OmniAuth::Strategy, which OmniAuth::Strategies::OAuth2 includes/inherits from.
+        # NOTE: When the callback_host option is not provided as an option, this will fallback to the orignal definition via super.
+        # NOTE: By default, Uber Omniauth will try and send the callback back to requester. This is a problem when the link on our marketing site is the requester, but the API needs to be the receiver. This allows us to retarget.
+        # NOTE: A callback_path should always be passed as an option, even if a callback_host is not.
+
+        callback_host = options[:callback_host]
```

Comment:
sorry, should have been more specific i was saying as a one-liner `if callback_host = options[:callback_host]`